core/src/main/java/ai/timefold/solver/core/impl/domain/variable/listener/support/VariableListenerSupport.java

```diff
@@ -33,12 +36,26 @@ public static <Solution_> VariableListenerSupport<Solution_> create(InnerScoreDi
     private final NotifiableRegistry<Solution_> notifiableRegistry;
     private final Map<Demand<?>, SupplyWithDemandCount> supplyMap = new HashMap<>();
 
+    private final List<ListVariableEvent> listVariableEventList;
+    private final ListVariableDescriptor<Solution_> listVariableDescriptor;
+    private final List<CascadingUpdateShadowVariableDescriptor<Solution_>> cascadingUpdateShadowVarDescriptorList;
+
     private boolean notificationQueuesAreEmpty = true;
     private int nextGlobalOrder = 0;
 
     VariableListenerSupport(InnerScoreDirector<Solution_, ?> scoreDirector, NotifiableRegistry<Solution_> notifiableRegistry) {
         this.scoreDirector = scoreDirector;
         this.notifiableRegistry = notifiableRegistry;
+        this.cascadingUpdateShadowVarDescriptorList = scoreDirector.getSolutionDescriptor().getEntityDescriptors().stream()
+                .flatMap(e -> e.getDeclaredCascadingUpdateShadowVariableDescriptors().stream())
+                .toList();
+        this.listVariableDescriptor = scoreDirector.getSolutionDescriptor().getEntityDescriptors().stream()
+                .flatMap(e -> e.getGenuineVariableDescriptorList().stream())
+                .filter(VariableDescriptor::isListVariable)
+                .map(d -> (ListVariableDescriptor<Solution_>) d)
+                .findFirst()
+                .orElse(null);
+        this.listVariableEventList = new LinkedList<>();
```

Comment:
Please don't use `LinkedList`.
https://kjellkod.wordpress.com/2012/02/25/why-you-should-never-ever-ever-use-linked-list-in-your-code-again/

I know that it has theoretical benefits in computer science, but in computer reality, they are no longer valid.